js/analytics.js

```diff
@@ -1,32 +1,81 @@
 import ReactGA from "react-ga";
 
-export default {
-  initialize: function () {
-    var _dntStatus = navigator.doNotTrack || navigator.msDoNotTrack;
-    var fxMatch = navigator.userAgent.match(/Firefox\/(\d+)/);
-    var ie10Match = navigator.userAgent.match(/MSIE 10/i);
-    var w8Match = navigator.appVersion.match(/Windows NT 6.2/);
-
-    if (fxMatch && Number(fxMatch[1]) < 32) {
-      _dntStatus = `Unspecified`;
-    } else if (ie10Match && w8Match) {
-      _dntStatus = `Unspecified`;
-    } else {
-      _dntStatus =
-        { "0": `Disabled`, "1": `Enabled` }[_dntStatus] || `Unspecified`;
-    }
+let Analytics = {};
+
+if (typeof window === `undefined`) {
+  // script is being run server-side
+
+  // A no-operation object with the same API surface as ReactGA
+  Analytics.ReactGA = {
+    initialize: () => {},
+    event: () => {},
+  };
+  Analytics.initialize = () => {};
+} else {
+  // script is being run client-side
+
```

Comment:
I'd recommend restructuring the script a little so that we don't even build the real analytics object unless DNT is off, with the DNT checking function made a top level function:

```js
function checkDoNotTrack() {
  // Treat server-side prerender as "do not track". If JS is enabled, the
  // React app will load client-side and add in GA later if permitted.
  if (typeof window === `undefined`) return true;

  // ...rest of code here...
}

// Are we allowed to track?
if (checkDoNotTrack() === false) {
  // build our analytics export
  const Analytics = {
    ReactGA,
    initialize: function() {
      // ...
    }
  };

  export default Analytics;
}

// If we're not allowed to track, return a "no-op" API object.
else {
  function noop() {};
  
  const NoAnalytics = {
    ReactGA: {
      initialize: noop,
      event: noop,
    },
    initialize: noop
  };

  export default NoAnalytics
}
```

Reply:
Nice suggestion!